pkg/mcp/sink/server_sink_test.go

```diff
@@ -147,3 +151,38 @@ func TestServerSink(t *testing.T) {
 		t.Fatalf("Connection should have failed: got %v want %v", err, nil)
 	}
 }
+
+func TestServerSinkRateLimitter(t *testing.T) {
```

Comment:
+1 for adding a test for this case. 

The current code relies on timing which can lead to flaky tests. One option (whitebox testing) would be to make this deterministic with a mock rate limiting that satisfies a common interfaces with the real rate limiter. You could then trigger N number of simultaneous connections and explicitly "allow" via the fake limiter. Another option (blackbox) would be to statistically verify connections were rate limiting over a period of time, similar to the previous MCP server NACK test. I think whitebox would be simpler for a unit test here. The blackbox test may be more approproate for an overall MCP stress test which I'm working on.

Reply:
>The current code relies on timing which can lead to flaky tests
I totally agree, I was sceptical of testing with a real rate limiter. I was initially considering the first option you mentioned above (whitebox testing) but I was concerned that it's going to be a bit of overkill, I will update to use a fake limiter.